backend/api/EventHandlers/MqttEventHandler.cs

```diff
@@ -55,35 +59,152 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
             await stoppingToken;
         }
 
-        private async void OnIsarConnect(object? sender, MqttReceivedArgs mqttArgs)
+        private async void OnIsarAvailability(object? sender, MqttReceivedArgs mqttArgs)
         {
-            var isarRobot = (IsarConnectMessage)mqttArgs.Message;
-            var robot = await RobotService.ReadByName(isarRobot.RobotId);
+            var isarAvailability = (IsarAvailabilityMessage)mqttArgs.Message;
+            var robot = await RobotService.ReadByName(isarAvailability.RobotName);
+
             if (robot == null)
             {
-                _logger.LogError(
-                    "ISAR instance for robot {name} connected. Could not find robot {name} in the database.",
-                    isarRobot.RobotId,
-                    isarRobot.RobotId
+                _logger.LogInformation(
+                    "Received message from new ISAR instance with robot name {name}. Adding new robot to database.",
+                    isarAvailability.RobotName
                 );
-                return;
+
+                var robotQuery = new CreateRobotQuery()
+                {
+                    Name = isarAvailability.RobotName,
+                    Model = RobotModel.Turtlebot,
```

Comment:
The robot model should not be hard coded, should come from the robot or some other place?

Reply:
Solved by including robot model in ISAR as an enum which is then translated to an enum in Flotilla.